ably/realtime_channel_spec_test.go

```diff
@@ -155,6 +157,1258 @@ func attachAndWait(t *testing.T, channel *ably.RealtimeChannel) {
+		if expected, got := ably.ChannelStateAttached, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+	})
+
+	t.Run("RTL4j1: AttachResume should be True when Attached", func(t *testing.T) {
+		t.Parallel()
+
+		app, err := ablytest.NewSandbox(nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer safeclose(t, app)
+
+		c, close := TransitionConn(t, nil, app.Options()...)
+		defer safeclose(t, close)
+
+		close = c.To(
+			connecting,
+			connected,
+		)
+
+		defer safeclose(t, close)
+
+		channelTransitioner := c.Channel("test")
+		channel := channelTransitioner.Channel
+
+		channelStateChanges := make(ably.ChannelStateChanges, 10)
+		channel.OnAll(channelStateChanges.Receive)
+
+		// channel state is initialized
+		if expected, got := ably.ChannelStateInitialized, channel.State(); expected != got {
+			t.Fatalf("expected %v; got %v", expected, got)
+		}
+
+		if channel.GetAttachResume() != false {
+			t.Fatalf("Channel attach resume should be false when channel state is INITIALIZED")
+		}
+
+		channelTransitioner.ToSpecifiedState(chAttaching, chAttached)
+
+		var channelStatechange ably.ChannelStateChange
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttaching, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttached, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+		if channel.GetAttachResume() != true {
+			t.Fatalf("Channel attach resume should be true when channel state is ATTCHED")
+		}
+
+		channelTransitioner.ToSpecifiedState(chDetaching)
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateDetaching, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		if channel.GetAttachResume() != false {
+			t.Fatalf("Channel attach resume should be false when channel state is DETACHING")
+		}
+
+		channelTransitioner.ToSpecifiedState(chFailed)
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateFailed, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		if channel.GetAttachResume() != false {
+			t.Fatalf("Channel attach resume should be false when channel state is FAILED")
+		}
+	})
+
+	t.Skip("RTL4j2: Rewind flag should allow to receive historic messages")
+	t.Run("RTL4j2: Rewind flag should allow to receive historic messages", func(t *testing.T) {
```

Comment:
@tcard , this test is having a flaky behaviour, I went through the code multiple times, seems to be no issue with the logic. Still, not sure why we don't receive messages. It will be great if you can debug and check what might be the real issue here

Reply:
Try using `persisted:test` as channel name. There are no history guarantees for non-persisted channels. Maybe that's the issue.

I'd also add a check that the ATTACH message carries the ATTACH_RESUME flag.

Other than that, I really don't know. This looks to me as if it's following RTL4j2 correctly. (I'm not sure why we should be testing this anyway, given it's mainly server behavior.) Maybe ask the realtime team?

Reply:
Tried, but not working. Maybe, you can try running test on your machine locally.

Reply:
It's a weird flaky behaviour at the same time, it's less flaky in debug mode ( like it passes most of the times in debug mode)

Reply:
I have marked test as skipped, you can take a look at it, if you get enough time

Reply:
I'd rather just leave it this way for now.